test/distributed/tensor/parallel/test_fsdp_2d_parallel.py

```diff
@@ -327,6 +343,78 @@ def test_2d_fsdp_state_enable_extension(self):
         fsdp_state = _get_module_fsdp_state(model)
         self.assertEqual(fsdp_state._enable_extension, True)
 
+    def _test_2d_e2e_training(
```

Comment:
Can we consolidate with L194? Feel free to refactor the code as you feel necessary.